### ISSUE TITLE
[QA] drawio fails to register mime type for .drawio files

### ISSUE THREAD

**Issue by jnweiger** [owncloud/drawio#29]
Seen with core 10.13.0-beta.1 and drawio-1.0.0 from marketplace

* the app installs  and registers in the [+] button to create new drawings. OK
* The drawing can be edited and saved just fine. OK
![image](https://github.com/owncloud/drawio/assets/1108546/fde470e7-6eb8-41de-a640-341d39a63346)

* Once saved, the app never again opens its editor. *.drawio files can only be downloaded
![image](https://github.com/owncloud/drawio/assets/1108546/108d1e7f-850d-4d38-ad41-56098c1e4283)
![image](https://github.com/owncloud/drawio/assets/1108546/39920cb7-3dea-4d36-9c70-dcf8960c5ce6)

**Comment by jnweiger** [comment]
-> mabye for #22

**Comment by jnweiger** [comment]
* maybe related? drawio fails with new web 7.0.3 UI only a blank page is shown
![image](https://github.com/owncloud/drawio/assets/1108546/ddd01b68-4358-4a01-a68a-43f60d8def70)

**Comment by hodyroff** [comment]
You sure this was with 1.0 from marketplace? It looks like the behavior from before that fix - looks like a regression to me, should be fixed?!

**Comment by pako81** [comment]
Reproducible in 10.13.0 beta 1 + drawio 1.0.0. Not reproducible in 10.12.2 + drawio 1.0.0. Must be a core issue.

**Comment by pako81** [comment]
Most likely related to https://github.com/owncloud/core/pull/40843

**Comment by pako81** [comment]
Removing https://github.com/owncloud/core/blob/v10.13.0-beta.1/resources/config/mimetypemapping.dist.json#L42 makes mimetype to be again correctly recognized.

**Comment by pako81** [comment]
> @jvillafanez so probably change mimetype to application/x-drawio in https://github.com/owncloud/drawio/blob/v1.0.0/js/files.js#L56 and https://github.com/owncloud/drawio/blob/v1.0.0/js/files.js#L115 ?

This seems to solve the issue in my 10.13.0-beta1 install.

**Comment by jvillafanez** [comment]
Potential patch:
```
diff --git a/js/files.js b/js/files.js
index c5b24cd..c00abba 100644
--- a/js/files.js
+++ b/js/files.js
@@ -53,7 +53,8 @@
                        // Attach to these two mimes
                        var mimes = {
                                "xml": {"mime": "application/xml", "type": "text"},
-                               "drawio": {"mime": "application/octet-stream", "type": "text"}
+                               "drawio": {"mime": "application/x-drawio", "type": "text"},
+                               "drawio_fallback": {"mime": "application/octet-stream", "type": "text"}
                        };
 
                        $.each(mimes, function (ext, attr) {
@@ -108,13 +109,12 @@
 $(document).ready(function () {
 
        OCA_DrawIO_Change_Icons = function () {
+               var mimeTypes = ['application/octet-stream', 'application/x-drawio'];
                $("#filestable")
                        .find("tr[data-type=file]")
                        .each(function () {
-                               if (($(this)
-                                       .attr("data-mime") === "application/octet-stream") && ($(this)
-                                       .find("div.thumbnail")
-                                       .length > 0)
+                               if (mimeTypes.indexOf($(this).attr('data-mime')) >= 0
+                               && ($(this).find("div.thumbnail").length > 0)
                                && $(this).find("span.extension").text() === '.drawio') {
                                        if ($(this)
                                                .find("div.thumbnail")

```
That should keep the app compatible with previous ownCloud versions.

I'm not sure if I'm missing something, but the file icon doesn't update by itself after saving, and you need to reload the page to get the drawio icon

**Comment by jvillafanez** [comment]
I don't think I have permissions to push to this repo, so someone else should open and merge the PR with the patch above (or something better)

**Comment by pako81** [comment]
The patch seems how to break with previous core versions, 10.12.2 i.e.

**Comment by pako81** [comment]
This seems to make it:

```
OCA.Drawio.FileList = {
                attach: function (fileList) {
                        // Don't attach to trashbin list
                        if (fileList.id === "trashbin") {
                                return;
                        }

                        // Attach to these three mimes
                        var mimes = [
                              {
                                "ext": "xml",
                                "mime": "application/xml",
                                "type": "text"
                              },
                              {
                                "ext": "drawio",
                                "mime": "application/x-drawio",
                                "type": "text"
                              },
                              {
                                "ext": "drawio",
                                "mime": "application/octet-stream",
                                "type": "text"
                              }
                        ];

                        $.each(mimes, function (index, object) {
                                fileList.fileActions.registerAction({
                                        name: "drawio-editor",
                                        displayName: t(OCA.Drawio.AppName, "Open in Draw.io"),
                                        mime: object.mime,
                                        permissions: OC.PERMISSION_READ | OC.PERMISSION_UPDATE,
                                        icon: function () {
                                                return OC.imagePath(OCA.Drawio.AppName, "drawio");
                                        },
                                        actionHandler: function (fileName, context) {
                                                var dir = fileList.getCurrentDirectory();
                                                OCA.Drawio.EditFileNewWindow(OC.joinPaths(dir, fileName));
                                        }
                                });
                                if(object.ext === 'drawio') {
                                        // Default for xml
                                        fileList.fileActions.setDefault(object.mime, "drawio-editor");
                                }
                        });

                }
        };

**Comment by pako81** [comment]
Proposed fix: https://github.com/owncloud/drawio/pull/30

**Comment by jnweiger** [comment]
Confirmed fixed via #30 in drawio-1.0.1 as bundled in core-10.13.0-rc.2
Followup issue #31
